File: src/main/java/org/minturtle/careersupport/interview/entity/InterviewMessage.java
package org.minturtle.careersupport.interview.entity;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;


@NoArgsConstructor
@AllArgsConstructor
@Builder
@Getter
@Document(collection = "interview_messages")
public class InterviewMessage {

    @Id
    private String id;

    private String templateId;

    private SenderType sender;

    private String content;

    @Builder.Default
    private Instant createdAt = Instant.now();

    public enum SenderType{
        USER, INTERVIEWER
    }
}

File: src/main/java/org/minturtle/careersupport/interview/entity/InterviewMessage.kt
package org.minturtle.careersupport.interview.entity

import org.minturtle.careersupport.common.utils.NanoIdGenerator
import org.springframework.data.annotation.Id
import java.time.Instant

class InterviewMessage(
    @Id
    var id: String = NanoIdGenerator.createNanoId(idSize = 10),
    val templateId: String,
    val sender: SenderType,
    val content: String,
    val createdAt: Instant = Instant.now()

) {

    enum class SenderType {
        USER,
        INTERVIEWER
    }

}